pkgs/development/python-modules/polars/default.nix

```diff
@@ -2,98 +2,304 @@
+
+  maturinBuildFlags = [
+    "-m"
+    "py-polars/Cargo.toml"
+  ];
 
-  pythonImportsCheck = [ "polars" ];
+  postInstall = ''
+    # Move polars.abi3.so -> polars.so
+    local polarsSo=""
+    local soName=""
+    while IFS= read -r -d "" p ; do
+      polarsSo=$p
+      soName="$(basename "$polarsSo")"
+      [[ "$soName" == polars.so ]] && break
+    done < <( find "$out" -iname "polars*.so" -print0 )
+    [[ -z "''${polarsSo:-}" ]] && echo "polars.so not found" >&2 && exit 1
+    if [[ "$soName" != polars.so ]] ; then
+      mv "$polarsSo" "$(dirname "$polarsSo")/polars.so"
+    fi
+  '';
+
+  pythonImportsCheck = [
+    "polars"
+  ];
+
+  passthru.tests.dynloading-1 =
+    runCommand "polars-dynloading-1"
+      {
+        nativeBuildInputs = [
+          (python.withPackages (ps: [
+            ps.pyarrow
+            polars
+          ]))
+        ];
+      }
+      ''
+        ((LD_DEBUG=libs python) |& tee $out | tail) << \EOF
+        import pyarrow
+        import polars
+        EOF
+        touch $out
+      '';
+  passthru.tests.dynloading-2 =
+    runCommand "polars-dynloading-2"
+      {
+        nativeBuildInputs = [
+          (python.withPackages (ps: [
+            ps.pyarrow
+            polars
+          ]))
+        ];
+        failureHook = ''
+          sed "s/^/    /" $out >&2
+        '';
+      }
+      ''
+        ((LD_DEBUG=libs python) |& tee $out | tail) << \EOF
+        import polars
+        import pyarrow
+        EOF
+      '';
+  passthru.tests.pytest = stdenv.mkDerivation {
+    pname = "${polars.pname}-pytest";
+
+    inherit (polars) version src;
+
+    requiredSystemFeatures = [ "big-parallel" ];
+
+    sourceRoot = "source/py-polars";
+    postPatch = ''
+      for f in * ; do
+        [[ "$f" == "tests" ]] || \
+        [[ "$f" == "pyproject.toml" ]] || \
+        rm -rf "$f"
+      done
+      for pat in "__pycache__" "*.pyc" ; do
+        find -iname "$pat" -exec rm "{}" ";"
+      done
+    '';
+    dontConfigure = true;
+    dontBuild = true;
+
+    doCheck = true;
+    checkPhase = "pytestCheckPhase";
+    nativeBuildInputs = [
+      (python.withPackages (ps: [
+        polars
+        ps.aiosqlite
+        ps.altair
+        ps.boto3
+        ps.deltalake
+        ps.flask
+        ps.flask-cors
+        ps.fsspec
+        ps.gevent
+        ps.hypothesis
+        ps.jax
+        ps.jaxlib
+        (ps.kuzu or null)
+        ps.moto
+        ps.nest-asyncio
+        ps.numpy
+        ps.openpyxl
+        ps.pandas
+        ps.pyarrow
+        ps.pydantic
+        (ps.pyiceberg or null)
+        ps.sqlalchemy
+        ps.torch
+        ps.xlsx2csv
+        ps.xlsxwriter
+        ps.zstandard
+        ps.cloudpickle
+      ]))
+    ];
+    nativeCheckInputs = [
+      pytestCheckHook
+      pytest-codspeed
+      pytest-cov
+      pytest-xdist
+      pytest-benchmark
+    ];
+
+    pytestFlagsArray = [
+      "-n auto"
+      "--dist loadgroup"
+      ''-m "slow or not slow"''
```

Comment:
May I ask about the purpose of specifying  `slow or not slow` test marks? Doesn't that equivalent to not specifying anything.

Reply:
Alright, I'll remove it during the tree-wide deprecation of the `pytestFlagsArray`.